modules/host/storageobligations.go

```diff
@@ -775,6 +782,19 @@ func (h *Host) threadedHandleActionItem(soid types.FileContractID) {
 			}
 			return
 		}
+		// If the obligation has no sector roots, we can remove the obligation and not
+		// submit a storage proof. The host payout for a failed empty contract
+		// includes the contract cost and locked collateral.
+		if len(so.SectorRoots) == 0 {
```

Comment:
this is clearer as an `else if` attached to the previous `if`, to make it clear that they're mutually exclusive